### PR TITLE
Better typechecking for micropip

### PR DESCRIPTION
**Pull request by hoodmane**
#### Better typechecking for micropip

I turned wheel and transaction into dataclasses so we can typecheck
their fields. I also went through and ensured that all functions have
typed arguments.

### CODE REVIEW

File: packages/micropip/test_micropip.py
Comment (ryanking13) [inline packages/micropip/test_micropip.py:60]: If this is added for debugging, please remove this before merging.

**Review by ryanking13 — APPROVED** [review]
It looks great to me. Thanks, @hoomane!

File: packages/micropip/test_micropip.py
Comment (hoodmane) [reply packages/micropip/test_micropip.py:60]: Oops, thanks for catching it.

**Comment by hoodmane** [comment]
Thanks for the review @ryanking13!

**Comment by rth** [comment]
Yes, thanks a lot @ryanking13 ! I'm happy that the project became a bit more sustainable with more active maintainers!

**Pull request merged**
